GatherPress/gatherpress
# New customisable rewrite bases for post types & taxonomies

### Description of the Change

This PR introduces a new settings section under GatherPress’ general settings to customise the rewrite slug for public post types and taxonomies.

If nothing is set, a localised string will be used, that is sanitised for use as part of an URL.

![Screenshot with localised German defaults.](https://github.com/user-attachments/assets/efe67b31-a7c3-4f77-b5e2-b9c84fe19eda)
_Screenshot with localised German defaults._



Closes #722

### How to test the Change


1. Go to _Events > Settings > General_
2. Change the rewrite slug for events
3. **Save** settings
4. Go to _Events > All Events_
5. Click on the **View** quick edit link
6. Confirm you opened your event under its new rewrite base, without any 404 errors.

### Changelog Entry

> Added - New customisable rewrite bases for post types & taxonomies

### Credits

Props @carstingaxion 

### Checklist:
